rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMNodeData.java

```diff
@@ -91,31 +95,53 @@
     private String jvmName;
 
+    private boolean locked;
+
+    private Client lockedBy;
+
+    private long lockTime;
+
+    /** Command line of deploying node. For RMDeployingNode only */
+    private String commandLine;
+
+    /** Description of a deploying or lost node. For RMDeployingNode only */
+    private String description;
+
     /**
      * Create an instance of {@link RMNodeData} and populate its fields with the content of a node.
      * @param rmNode the object to take the values from
      * @return a new instance of {@link RMNodeData} with the values of the given {@link RMNode}
      */
     public static RMNodeData createRMNodeData(RMNode rmNode) {
-        RMNodeData rmNodeData = new RMNodeData(rmNode.getNodeName(),
-                                               rmNode.getNodeURL(),
-                                               rmNode.getOwner(),
-                                               rmNode.getProvider(),
-                                               rmNode.getUserPermission(),
-                                               rmNode.getState(),
-                                               rmNode.getStateChangeTime(),
-                                               rmNode.getHostName(),
-                                               rmNode.getJmxUrls(),
-                                               rmNode.getDescriptorVMName());
-        return rmNodeData;
+        Builder builder = new Builder().name(rmNode.getNodeName())
+                                       .nodeUrl(rmNode.getNodeURL())
+                                       .provider(rmNode.getProvider())
+                                       .state(rmNode.getState())
+                                       .stateChangeTime(rmNode.getStateChangeTime())
+                                       .locked(rmNode.isLocked())
+                                       .lockedBy(rmNode.getLockedBy())
+                                       .lockTime(rmNode.getLockTime());
+        if (!rmNode.getState().equals(NodeState.DEPLOYING) && !rmNode.getState().equals(NodeState.LOST)) {
+            builder.owner(rmNode.getOwner())
+                   .userPermission(rmNode.getUserPermission())
+                   .hostname(rmNode.getHostName())
+                   .jmxUrls(rmNode.getJmxUrls())
+                   .jvmName(rmNode.getDescriptorVMName());
+        } else {
+            builder.commandLine(((RMDeployingNode) rmNode).getCommandLine())
+                   .description(((RMDeployingNode) rmNode).getDescription());
+        }
+
+        return builder.build();
     }
 
     public RMNodeData() {
         // Required empty constructor
     }
 
-    public RMNodeData(String name, String nodeUrl, Client owner, Client provider, Permission permission,
-            NodeState state, long stateChangeTime, String hostName, String[] jmxUrls, String jvmName) {
+    private RMNodeData(String name, String nodeUrl, Client owner, Client provider, Permission permission,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 15 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)